apps/central-scan/frontend/src/app_root.tsx

```diff
@@ -476,19 +476,35 @@ export function AppRoot({
     }
   }, [electionJustLoaded, usbDrive.status]);
 
+  // We check configureMutation's status and stop configuration attempts after a failed attempt.
+  // This prevents the app from looping with continued failed configurations.
+  // If the user ejects the USB drive, we reset configureMutation's status so configuration
+  // may be attempted when a new or modified USB drive is inserted.
+  useEffect(() => {
+    if (
+      (configureMutation.isError || configureMutation?.data?.err()) &&
+      usbDrive.status === 'ejected'
+    ) {
+      configureMutation.reset();
+    }
+  }, [usbDrive.status, configureMutation]);
+
```

Comment:
I'm not sure if this is applicable here, but it might be worth looking at how we approach this in VxScan: https://github.com/votingworks/vxsuite/blob/b5cc9385dfb37791f9b3d0142915f6d93e65f584/apps/scan/frontend/src/screens/unconfigured_election_screen_wrapper.tsx#L26-L33.

We have a listener on the unconfigured screen to check if the USB status changes to `mounted` and only configure in that case. I'm pretty sure it has the correct behavior.

Reply:
Discussed on Slack: the best long term fix is to move `central-scan` over to the newer `@votingworks/usb-drive` lib and get USB drive status from the backend. This will allow us to match the pattern in `scan`.

If we want to make a shorter fix to similar effect, we can use `useExternalStateChangeListener` as a stopgap.

For now we'll merge this to the m17 branch so at least some fix is in.